code/modules/mob/mob.dm

```diff
@@ -1098,25 +1098,23 @@ var/list/slot_equipment_priority = list( \
 				if(!L.key)
 					creatures += L
 		var/picked = input("Please select an NPC to respawn as", "Respawn as NPC")  as null|anything in creatures
+		var/mob/dead/observer/O = src
 		switch(picked)
 			if("Mouse")
-				GLOB.respawnable_list -= usr
-				become_mouse()
-				spawn(5)
-					GLOB.respawnable_list += usr
+				become_mouse(O)
 			else
 				var/mob/living/NPC = picked
 				if(istype(NPC) && !NPC.key)
-					GLOB.respawnable_list -= usr
-					NPC.key = key
-					spawn(5)
-						GLOB.respawnable_list += usr
+					if(mind)
```

Comment:
There's multiple thing wrong with this.

Antagonist status and traitor memory, objectives etc. is stored in the mind.

This means:

![image](https://user-images.githubusercontent.com/40092670/65896423-34555180-e37b-11e9-9675-007ea55eec2c.png)


I think it's self-evident what goes wrong

Not to mention you never removed them from the respawnable_list which is going to be problematic when some proc look for a respawnable and see someone who is already a mouse.

Remove this fix, this creates more problems than it solve.

Reply:
Fox told me that you should never destroy the mind of a ghost of you can help it, when I used code based on this very function in the diona nymph thing. So why is it acceptable here and not there? Obviously mice keeping antag objectives is a problem, but given the circumstance I'm pretty sure my diona pods would do the same thing, and that would be even worse.

Reply:
That's something that I cannot comment on as I do not know the specific of this PR. All I know is that this fix create more problems than it solve.

In fact, I don't think this even solve any issue at all. This one should be removed. You should avoid refactoring things you do not understand. Even I am not confident in refactoring datum, and you, for one, did not know that mind was responsible for storing antag status and abilities.